ludwig/schema/encoders.py

```diff
@@ -0,0 +1,134 @@
+    "Class variable pointing to the corresponding Encoder class."
+
+    encoder: str
+    "Name corresponding to an encoder."
+
+
+@dataclass
+class DenseEncoder(Schema):
+    """DenseEncoder is a dataclass that configures the parameters used for a dense encoder."""
+
+    encoder_class: ClassVar[Encoder] = DenseEncoder
+
+    type: str = "dense"
+
+    fc_layers: Optional[List[dict]] = schema_utils.DictList(
+        default=None,
+        description="List of fully connected layers to use in the encoder.",
+    )
+
+    num_layers: Optional[int] = schema_utils.PositiveInteger(
+        default=1,
+        description="Number of stacked fully connected layers that the input to the feature passes through.",
+    )
+
+    output_size: Optional[int] = schema_utils.PositiveInteger(
+        default=256,
+        description="Size of the output of the feature.",
+    )
+
+    use_bias: Optional[bool] = schema_utils.Boolean(
+        default=True,
+        description="Whether the layer uses a bias vector.",
+    )
+
+    weights_initializer: Optional[Union[str, dict]] = schema_utils.StringOptions(  # TODO: Add support for String/Dict
+        [
+            "constant",
+            "identity",
+            "zeros",
+            "ones",
+            "orthogonal",
+            "normal",
+            "uniform",
+            "truncated_normal",
+            "variance_scaling",
+            "glorot_normal",
+            "glorot_uniform",
+            "xavier_normal",
+            "xavier_uniform",
+            "he_normal",
+            "he_uniform",
+            "lecun_normal",
+            "lecun_uniform",
+        ],
+        default="glorot_uniform",
+        description="Initializer for the weight matrix.",
+    )
+
+    bias_initializer: Optional[Union[str, dict]] = schema_utils.StringOptions(  # TODO: Add support for String/Dict
+        [
+            "constant",
+            "identity",
+            "zeros",
+            "ones",
+            "orthogonal",
+            "normal",
+            "uniform",
+            "truncated_normal",
+            "variance_scaling",
+            "glorot_normal",
+            "glorot_uniform",
+            "xavier_normal",
+            "xavier_uniform",
+            "he_normal",
+            "he_uniform",
+            "lecun_normal",
+            "lecun_uniform",
+        ],
+        default="zeros",
+        description="Initializer for the bias vector.",
+    )
+
+    norm: Optional[Union[str]] = schema_utils.StringOptions(
+        ["batch", "layer"],
+        allow_none=True,
+        default=None,
+        description="Normalization to use in the dense layer.",
+    )
+
+    norm_params: Optional[dict] = schema_utils.Dict(
+        default=None,
+        description="Parameters for normalization if norm is either batch or layer.",
+    )
+
+    activation: Optional[str] = schema_utils.StringOptions(
+        ["elu", "leakyRelu", "logSigmoid", "relu", "sigmoid", "tanh", "softmax"],
```

Comment:
Could this be replaced with `activations.keys()` (perhaps renamed to `activations_registry`) from ludwig.utils.torch_utils?

Reply:
Same as above, will include in the encoder schema refactor.

Reply:
Going to remove this file from this PR since it is not relevant until the encoder schema PR.